sdk/vision/azure-ai-vision-imageanalysis/README.md

```diff
@@ -0,0 +1,340 @@
+
+The `ImageAnalysisClient` has two overloads for the method `analyze`:
+* Analyze an image from a memory buffer, using the [BinaryData](https://learn.microsoft.com/java/api/com.azure.core.util.binarydata) object. The client will upload the image to the service as the REQUEST body. 
+* Analyze an image from a publicly-accessible URL, using the [java.lang.URL](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/net/URL.html) object. The client will send the image URL to the service. The service will fetch the image.
+
+The examples below show how to do both. The `analyze` examples populate the input `BinaryData` object by loading an image from a file on disk.
+
+### Supported image formats
+
+Image Analysis works on images that meet the following requirements:
+* The image must be presented in JPEG, PNG, GIF, BMP, WEBP, ICO, TIFF, or MPO format
+* The file size of the image must be less than 20 megabytes (MB)
+* The dimensions of the image must be greater than 50 x 50 pixels and less than 16,000 x 16,000 pixels
+
+
+## Examples
+
+The following sections provide code snippets covering these common Image Analysis scenarios:
+
+* [Generate an image caption for an image file](#generate-an-image-caption-for-an-image-file)
+* [Generate an image caption for an image URL](#generate-an-image-caption-for-an-image-url)
+* [Extract text (OCR) from an image file](#extract-text-from-an-image-file)
+* [Extract text (OCR) from an image URL](#extract-text-from-an-image-url)
+
+These snippets use the synchronous `client` from [Create and authenticate the client](#3-create-and-authenticate-the-client).
+
+See the [Samples][samples] folder for fully working samples for all visual features, including asynchronous clients.
+
+### Generate an image caption for an image file
+
+This example demonstrates how to generate a one-sentence caption for the image file `sample.jpg` using the `ImageAnalysisClient`. The synchronous (blocking) `analyze` method call returns an `ImageAnalysisResult` object.
+A call to `getCaption()` on this result will return a `CaptionResult` object.
+It contains the generated caption and its confidence score in the range [0, 1]. By default the caption may contain gender terms such as "man", "woman", or "boy", "girl". You have the option to request gender-neutral terms such as "person" or "child" by setting `genderNeutralCaption` to `true` when calling `analyze`, as shown in this example.
+
+Notes:
+* Caption is only available in some Azure regions. See [Prerequisites](#prerequisites).
+* Caption is only supported in English at the moment.
+
+```java imports-caption-file-snippet
+import com.azure.ai.vision.imageanalysis.ImageAnalysisOptionsBuilder;
+import com.azure.ai.vision.imageanalysis.models.ImageAnalysisResult;
+import com.azure.ai.vision.imageanalysis.models.VisualFeatures;
+import com.azure.core.util.BinaryData;
+import java.io.File;
+import java.util.Arrays;
+```
+```java caption-file-snippet
```

Comment:
I don't know if we want to include `import`s in our README code samples, I feel this can be left to the richer sample files.

@JonathanGiles I know you've done a lot of work this year around README documentation, is there any guidance around this?

Reply:
@alzimmermsft @JonathanGiles I see this README.md is a self-contained quick-start guide. My expectation is that developers can copy this code and run it quickly. I don't want them to try and figure out which packages each class comes from. Unless this violates some hard guideline, I'd like to keep it as is. Thanks!